execution/exchange/duplicate_label.go

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) The Thanos Community Authors.
+// Licensed under the Apache License 2.0.
+
+package exchange
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/prometheus/prometheus/model/labels"
+
+	"github.com/thanos-io/promql-engine/execution/model"
+	"github.com/thanos-io/promql-engine/extlabels"
+	"github.com/thanos-io/promql-engine/query"
+)
+
+type duplicateLabelCheckOperator struct {
+	model.OperatorTelemetry
+
+	once sync.Once
+	next model.VectorOperator
+
+	checkDuplicates func([]model.StepVector) error
```

Comment:
Shouldn't we run this function only if there is a collision in series?

Reply:
It is a noop if we detect one currently

Reply:
I mean if we detect that no collision can happen